pisces/data_sets.py

```diff
@@ -327,6 +334,440 @@ def find_data_sets(cls, root: str | Path) -> Dict[str, 'DataSetObject']:
+        new_col_name = f"{col}_smoothed" if not overwrite else col
+        df = df.with_columns(
+            pl.Series(gaussian_filter1d(df[col].to_numpy(), sigma)).alias(new_col_name)
+        )
+    return df
+
+
+def fill_gaps_in_accelerometer_data(acc: pl.DataFrame, smooth: bool = False, final_sampling_rate_hz: int | None = None) -> np.ndarray:
+    # median sampling rate (to account for missing data)
+    sampling_period_s = acc[acc.columns[0]].diff().median() # 1 / sampling_rate_hz
+    
+    # Step 0: Save the original 'timestamp' column as 'timestamp_raw'
+    acc_resampled = acc.with_columns(acc[acc.columns[0]].alias('timestamp'))
+
+    #TODO: Check non int sampling rates
+    # if isinstance(final_sampling_rate_hz, int):
+    final_rate_sec = 1 / final_sampling_rate_hz
+    print(f"resampling to {final_sampling_rate_hz}Hz ({final_rate_sec:0.5f}s) from {int(1/sampling_period_s)} Hz ({sampling_period_s:0.5f}s)")
+    # make a new data frame with the new timestamps
+    # do this using linear interpolation
+
+    median_time = acc_resampled['timestamp'].to_numpy()
+    final_timestamps = np.arange(median_time.min(), median_time.max() + final_rate_sec, final_rate_sec)
+    median_data = acc_resampled[:, 1:4].to_numpy()
+    new_data = np.zeros((final_timestamps.shape[0], median_data.shape[1]))
+    for i in range(median_data.shape[1]):
+        new_data[:, i] = np.interp(final_timestamps, median_time, median_data[:, i])
+    acc_resampled = pl.DataFrame({
+        'timestamp': final_timestamps, 
+        **{
+            acc_resampled.columns[i+1]: new_data[:, i] 
+            for i in range(new_data.shape[1])
+        }})
+
+    if smooth:
+        acc_resampled = apply_gausian_filter(acc_resampled, overwrite=True)
+
+    return acc_resampled
+
+# %% ../nbs/01_data_sets.ipynb 18
+class DataProcessor:
+    def __init__(self, 
+                 data_set: DataSetObject,
+                 model_input: ModelInput,
+                 output_feature: str='psg',
+                 output_type: ModelOutputType=ModelOutputType.WAKE_LIGHT_DEEP_REM,
+                 psg_type: PSGType=PSGType.HAS_N4,
+                 ):
+        self.data_set = data_set
+        self.input_features = model_input.input_features
+        self.output_feature = output_feature
+        self.output_type = output_type
+        self.psg_type = psg_type
+        self.model_input = model_input
+
+        if isinstance(model_input, ModelInput1D):
+            self.input_window_time = model_input.input_window_time
+            self.input_sampling_hz = model_input.input_sampling_hz
+            self.input_window_samples = model_input.input_window_samples
+            self.model_input_dimension = model_input.model_input_dimension
+        elif isinstance(model_input, ModelInputSpectrogram):
+            self.input_sampling_hz = model_input.input_sampling_hz
+            self.spectrogram_preprocessing_config = model_input.spectrogram_preprocessing_config
```

Comment:
I think there are 2 nice getters that could simplify some other code

```python
@property
def is_1D(self):
    return isinstance(self.model_input, ModelInput1D)

@property
def is_specgtrogram(self):
    isinstance(model_input, ModelInputSpectrogram)
```
then we can change the end of this to 
```python
if self.is_1D:
...
elif self.is_spectrogram:
...```
This makes the code more readable and less error-prone elsewhere, like in `preprocess_data_for_subject` later on.